Mathlib/Algebra/Category/Ring/FinitePresentation.lean

```diff
@@ -0,0 +1,171 @@
+/-
+Copyright (c) 2025 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang, Christian Merten
+-/
+import Mathlib.Algebra.Category.Ring.FilteredColimits
+import Mathlib.CategoryTheory.Limits.Preserves.Over
+import Mathlib.CategoryTheory.Limits.Shapes.FiniteMultiequalizer
+import Mathlib.CategoryTheory.Presentable.Finite
+import Mathlib.RingTheory.EssentialFiniteness
+import Mathlib.RingTheory.FinitePresentation
+
+/-!
+
+# Finitely presentable objects in `Under R` with `R : CommRingCat`
+
+In this file, we show that finitely presented algebras are finitely presentable in `Under R`,
+i.e. `Hom_R(S, -)` preserves filtered colimits.
+
+-/
+
+open CategoryTheory Limits
+
+universe vJ uJ u
+
+variable {J : Type uJ} [Category.{vJ} J] [IsFiltered J]
+variable (R : CommRingCat.{u}) (F : J ⥤ CommRingCat.{u}) (α : (Functor.const _).obj R ⟶ F)
+variable {S : CommRingCat.{u}} (f : R ⟶ S) (c : Cocone F) (hc : IsColimit c)
+variable [PreservesColimit F (forget CommRingCat)]
+
+include hc in
+/--
+Given a filtered diagram `F` of rings over `R`, `S` an (essentially) of finite type `R`-algebra,
+and two ring homs `a : S ⟶ Fᵢ` and `b : S ⟶ Fⱼ` over `R`.
+If `a` and `b` agree at `S ⟶ colimit F`,
+then there exists `k` such that `a` and `b` are equal at `S ⟶ F_k`.
+In other words, the map `colimᵢ Hom_R(S, Fᵢ) ⟶ Hom_R(S, colim F)` is injective.
+-/
+lemma exists_comp_map_eq_of_isColimit (hf : f.hom.EssFiniteType)
+    {i : J} (a : S ⟶ F.obj i) (ha : f ≫ a = α.app i)
+    {j : J} (b : S ⟶ F.obj j) (hb : f ≫ b = α.app j)
+    (hab : a ≫ c.ι.app i = b ≫ c.ι.app j) :
+    ∃ (k : J) (hik : i ⟶ k) (hjk : j ⟶ k),
+      a ≫ F.map hik = b ≫ F.map hjk := by
+  classical
+  obtain ⟨hc'⟩ := PreservesColimit.preserves (F := forget _) hc
+  choose k f₁ f₂ h using fun x : S ↦
+    (Types.FilteredColimit.isColimit_eq_iff _ hc').mp congr(($hab).hom x)
+  let D : MulticospanIndex J :=
```

Comment:
You may have to do a fix after merging with master because of https://github.com/leanprover-community/mathlib4/pull/20988